src/communication/websocket/auth.cpp

```diff
@@ -0,0 +1,25 @@
+// Copyright 2021 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include "communication/websocket/auth.hpp"
+
+namespace communication::websocket {
+
+bool SafeAuth::Authenticate(const std::string &username, const std::string &password) {
+  std::optional<auth::User> maybe_user;
+  {
+    // TODO change lock to readlock, auth now creates user...
```

Comment:
What about this?

Reply:
Well, this is more to do with this method from auth module:
https://github.com/memgraph/memgraph/blob/master/src/auth/auth.cpp#L70-L162
And the issue is I am not sure everything that I might affect by refactoring this, and not sure how much time would it take me to do it...

Reply:
No, it doesn't worth it now. I check the auth module, and I just don't like how it is written. Especially the `FLAGS_auth_module_create_missing_user` variable is super scary to me. I think that shouldn't be part of the module, because if that is true, then you cannot enforce the authentication without creating new user.

So right into my next question which is also meant for @antonio2368 and @kostasrim: do you think enabling a newly created user reading all the log messages is fine? I would say the user should have very high privileges in order to do this.

Reply:
I think it makes sense to add another role for log reading over websocket

Reply:
As far as I know, modules are a solution for a really specific problem we solved for one of our customers, so I wouldn't put too much thought into it now. All the flags seem really specific and have questionable side effects. We should protect access to logs with some kind of permission, that's for sure.

@jbajic can you try and add that as part of this PR?